language/tools/resource-viewer/src/lib.rs

```diff
@@ -300,7 +301,7 @@ impl RemoteCache for NullStateView {
         &self,
         _address: &AccountAddress,
         _tag: &StructTag,
-    ) -> PartialVMResult<Option<Vec<u8>>> {
+    ) -> PartialVMResult<Option<Cow<[u8]>>> {
```

Comment:
So I can see that by using `Cow` you are potentially saving a copy of the bytes, but I wonder if a simple reference would also achieve the same goal? IIRC there's nothing in the Move VM that actually wants to modify these bytes so the "on-write" part seem redundant to me.

Reply:
The problem is from `StateView`. The `StateView` returns a owned `Vec<u8>` on storage read. In some implementation of the `RemoteCache` there isn't this caching behavior so a `Cow` is actually needed over there.